File: base/src/main/java/dev/nathanpb/dml/enums/MatterType.java
package dev.nathanpb.dml.enums;

import dev.nathanpb.dml.dmlSimulacrum;
import dev.nathanpb.dml.item.ItemMatter;
import net.minecraft.item.Item;
import net.minecraft.util.registry.Registry;

public enum MatterType {

    OVERWORLD(Registry.ITEM.get(dmlSimulacrum.id("overworld_matter"))),
    HELLISH(Registry.ITEM.get(dmlSimulacrum.id("hellish_matter"))),
    EXTRATERRESTRIAL(Registry.ITEM.get(dmlSimulacrum.id("extraterrestrial_matter")));

    private final ItemMatter matter;

    MatterType(Item matter) {
        this.matter = (ItemMatter) matter;
    }

    public ItemMatter getItem() {
        return matter;
    }
}

File: base/src/main/kotlin/dev/nathanpb/dml/enums/MatterType.kt
package dev.nathanpb.dml.enums

import dev.nathanpb.dml.MOD_ID
import dev.nathanpb.dml.identifier
import dev.nathanpb.dml.utils.getItemFromRegistry
import net.minecraft.item.Item
import net.minecraft.text.Text
import net.minecraft.util.Formatting

/** simulacrum only! */
enum class MatterType(id: String, color: Formatting) {

    OVERWORLD("overworld", Formatting.GREEN),
    HELLISH("hellish", Formatting.RED),
    EXTRATERRESTRIAL("extraterrestrial", Formatting.LIGHT_PURPLE);


    val text: Text = Text.translatable("modelType.${MOD_ID}.${id}").formatted(color)

    val matter: Item? = getItemFromRegistry(identifier("${id}_matter"))

}